### DISCUSSION TITLE
Hi, hello. I found an error while building curl using OpenSSL cURL. I cannot bypass it and would like to seek help

### DISCUSSION THREAD

**Discussion by ghost** [jasonacox/Build-OpenSSL-cURL#73] [Q&A]
==> /tmp/curl-7.74.0-arm64.log <==
rm -f vauth/*.o
rm -f vauth/*.lo
rm -f vquic/*.o
rm -f vquic/*.lo
rm -f vssh/*.o
rm -f vssh/*.lo
rm -f vtls/*.o
rm -f vtls/*.lo
rm -f *.lo
Making clean in src
Making clean in ../docs
Making clean in .
test -z "curl.html curl-config.html mk-ca-bundle.html curl.pdf curl-config.pdf mk-ca-bundle.pdf curl.1.dist curl-config.1.dist curl.1" || rm -f curl.html curl-config.html mk-ca-bundle.html curl.pdf curl-config.pdf mk-ca-bundle.pdf curl.1.dist curl-config.1.dist curl.1
rm -rf .libs _libs
rm -f *.lo
Making clean in cmdline-opts
rm -rf .libs _libs
rm -f *.lo
 rm -f curl
test -z "tool_hugehelp.c" || rm -f tool_hugehelp.c
rm -rf .libs _libs
rm -rf ../lib/.libs ../lib/_libs
test -z "" || rm -f 
rm -f *.o
rm -f ../lib/*.o
rm -f ../lib/*.lo
rm -f *.lo
test -z "projects/Windows/VC6/lib/libcurl.dsp.dist projects/Windows/VC6/src/curl.dsp.dist projects/Windows/VC7/lib/libcurl.vcproj.dist projects/Windows/VC7/src/curl.vcproj.dist projects/Windows/VC7.1/lib/libcurl.vcproj.dist projects/Windows/VC7.1/src/curl.vcproj.dist projects/Windows/VC8/lib/libcurl.vcproj.dist projects/Windows/VC8/src/curl.vcproj.dist projects/Windows/VC9/lib/libcurl.vcproj.dist projects/Windows/VC9/src/curl.vcproj.dist projects/Windows/VC10/lib/libcurl.vcxproj.dist projects/Windows/VC10/src/curl.vcxproj.dist projects/Windows/VC11/lib/libcurl.vcxproj.dist projects/Windows/VC11/src/curl.vcxproj.dist projects/Windows/VC12/lib/libcurl.vcxproj.dist projects/Windows/VC12/src/curl.vcxproj.dist projects/Windows/VC14/lib/libcurl.vcxproj.dist projects/Windows/VC14/src/curl.vcxproj.dist projects/Windows/VC15/lib/libcurl.vcxproj.dist projects/Windows/VC15/src/curl.vcxproj.dist" || rm -f projects/Windows/VC6/lib/libcurl.dsp.dist projects/Windows/VC6/src/curl.dsp.dist projects/Windows/VC7/lib/libcurl.vcproj.dist projects/Windows/VC7/src/curl.vcproj.dist projects/Windows/VC7.1/lib/libcurl.vcproj.dist projects/Windows/VC7.1/src/curl.vcproj.dist projects/Windows/VC8/lib/libcurl.vcproj.dist projects/Windows/VC8/src/curl.vcproj.dist projects/Windows/VC9/lib/libcurl.vcproj.dist projects/Windows/VC9/src/curl.vcproj.dist projects/Windows/VC10/lib/libcurl.vcxproj.dist projects/Windows/VC10/src/curl.vcxproj.dist projects/Windows/VC11/lib/libcurl.vcxproj.dist projects/Windows/VC11/src/curl.vcxproj.dist projects/Windows/VC12/lib/libcurl.vcxproj.dist projects/Windows/VC12/src/curl.vcxproj.dist projects/Windows/VC14/lib/libcurl.vcxproj.dist projects/Windows/VC14/src/curl.vcxproj.dist projects/Windows/VC15/lib/libcurl.vcxproj.dist projects/Windows/VC15/src/curl.vcxproj.dist
rm -rf .libs _libs
rm -f *.lo

==> /tmp/curl-7.74.0-iOS-armv7-bitcode.log <==
configure: WARNING: zstd.h: present but cannot be compiled
configure: WARNING: zstd.h:     check for missing prerequisite headers?
configure: WARNING: zstd.h: see the Autoconf documentation
configure: WARNING: zstd.h:     section "Present But Cannot Be Compiled"
configure: WARNING: zstd.h: proceeding with the compiler's result
configure: WARNING:     ## ------------------------------------------------------------------ ##
configure: WARNING:     ## Report this to a suitable curl mailing list: https://curl.se/mail/ ##
configure: WARNING:     ## ------------------------------------------------------------------ ##
checking for zstd.h... no
checking for lber.h... no
checking for ldap.h... no
checking for ldapssl.h... no
checking for ldap_ssl.h... no
checking for LDAP libraries... cannot find LDAP libraries
configure: WARNING: Cannot find libraries for LDAP support: LDAP disabled
checking whether to enable IPv6... yes
checking if struct sockaddr_in6 has sin6_scope_id member... yes
checking if argv can be written to... no
configure: WARNING: the previous check could not be made default was used
checking if GSS-API support is requested... no
checking whether to enable Windows native SSL/TLS (Windows native builds only)... no
checking whether to enable Secure Transport... no
checking whether to enable Amiga native SSL/TLS (AmiSSL)... no
checking for HMAC_Update in -lcrypto... no
checking for HMAC_Init_ex in -lcrypto... no
checking OpenSSL linking with -ldl... no
checking OpenSSL linking with -ldl and -lpthread... no
configure: OPT_SSL: /Users/z/Desktop/curl/mmm/Build-OpenSSL-cURL/curl/../openssl/iOS
configure: OPENSSL_ENABLED: 
configure: error: --with-ssl was given but OpenSSL could not be detected

==> /tmp/curl-7.74.0-x86_64.log <==
checking for x86_64-apple-darwin-pkg-config... no
checking for pkg-config... /usr/local/bin/pkg-config
checking for zlib options with pkg-config... found
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
configure: found both libz and libz.h header
checking for BrotliDecoderDecompress in -lbrotlidec... no
checking brotli/decode.h usability... no
checking brotli/decode.h presence... no
checking for brotli/decode.h... no
checking for ZSTD_createDStream in -lzstd... yes
checking zstd.h usability... yes
checking zstd.h presence... yes
checking for zstd.h... yes
checking for lber.h... yes
checking for ldap.h... yes
checking for ldapssl.h... no
checking for ldap_ssl.h... no
checking for LDAP libraries... -lldap
checking for ldap_url_parse... yes
checking for ldap_init_fd... no
checking whether to enable IPv6... yes
checking if struct sockaddr_in6 has sin6_scope_id member... yes
checking if argv can be written to... yes
checking if GSS-API support is requested... no
checking whether to enable Windows native SSL/TLS (Windows native builds only)... no
checking whether to enable Secure Transport... no
checking whether to enable Amiga native SSL/TLS (AmiSSL)... no
configure: error: /Users/z/Desktop/Build-OpenSSL-cURL-master/curl/../openssl/Mac is a bad --with-ssl prefix!

==> /tmp/curl-8.7.1-x86_64.log <==
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libdefault-lib-cipher_chacha20_hw.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libdefault-lib-cipher_chacha20_poly1305_hw.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libdefault-lib-cipher_sm4_hw.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libdefault-lib-cipher_tdes_default_hw.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libdefault-lib-cipher_tdes_hw.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libdefault-lib-cipher_tdes_wrap_hw.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libdefault-lib-endecoder_common.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcrypto-lib-sm2_crypt.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcrypto-lib-sm2_sign.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcrypto-lib-sm2_key.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libdefault-lib-cipher_tdes_common.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcrypto-lib-wp_block.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcommon-lib-ciphercommon_ccm_hw.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcommon-lib-ciphercommon_hw.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcommon-lib-der_ec_gen.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcommon-lib-der_ecx_gen.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcommon-lib-der_rsa_gen.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcommon-lib-der_dsa_gen.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libdefault-lib-der_sm2_gen.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libdefault-lib-digest_to_nid.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcommon-lib-ciphercommon_gcm_hw.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
ld: warning: all bitcode will be dropped because '/Users/z/Downloads/Build-OpenSSL-cURL-master/openssl/Mac/lib/libcrypto.a(libcrypto-lib-siv128.o)' was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. 
Undefined symbols for architecture x86_64:
  "_SSL_get0_group_name", referenced from:
      _ossl_connect_common in libcurl.a(libcurl_la-openssl.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [curl] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all-recursive] Error 1

**Comment by jasonacox** [comment]
Hi @ZhichaoZhan 

> configure: error: --with-ssl was given but OpenSSL could not be detected

This indicates that OpenSSL was not built or was not built correctly. Did you use any `build.sh` command options (e.g. versions)?
